File: backend/app/core/study_language.py
import enum


class StudyLanguage(str, enum.Enum):
    """Study target language for a user account."""

    JAPANESE = "ja"
    CHINESE = "zh"


STUDY_LANGUAGES: tuple[StudyLanguage, ...] = (
    StudyLanguage.JAPANESE,
    StudyLanguage.CHINESE,
)

LANGUAGE_LABELS: dict[StudyLanguage, str] = {
    StudyLanguage.JAPANESE: "Japanese",
    StudyLanguage.CHINESE: "Chinese",
}

File: frontend/src/lib/study-language.ts
export type StudyLanguage = 'ja' | 'zh'

export const STUDY_LANGUAGE_LABELS: Record<StudyLanguage, string> = {
  ja: 'Japanese',
  zh: 'Chinese',
}

export function isJapanese(language: StudyLanguage): boolean {
  return language === 'ja'
}
